### PR TITLE
Chown with environment variables not set should fail

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/buildah/issues/3380

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
Still fighting the test system @rhatdan

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL
@nalind @vrothberg @TomSweeneyRedHat @ashley-cui PTAL

File: add.go [hunk 1]
```diff
@@ -602,12 +602,43 @@ func (b *Builder) userForRun(mountPoint string, userspec string) (specs.User, st
...
+
+	split := strings.SplitN(userspec, ":", 2)
+	user = split[0]
+	if len(split) > 1 {
+		group = split[1]
+	}
+
+	// If userspec did not specify any values for user or group, then fail
+	if user == "" && group == "" {
+		return 0, 0, errors.Errorf("can't find uid for user %s", userspec)
```
Comment (vrothberg) [inline add.go:619]: Just in case of a repush: can you change `%s` to `%q`.  That'll make empty strings clear in the logs.

**Review by vrothberg — COMMENTED** [review]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
@rhatdan I'll let you decide if you want to blast this in or follow up with a quick nit fix.

**Pull request merged**